File: src/MeanscriptCpp/pub/MSOutputArray.cpp
#include "MS.h"
namespace meanscript {
using namespace meanscriptcore;

MSOutputArray::MSOutputArray ()
{
	maxSize = globalConfig.outputArraySize;
	{ buffer.reset(maxSize); };
	index = 0;
}


void MSOutputArray::close () 
{
	index = -1;
}


void MSOutputArray:: writeByte (uint8_t b) 
{
	CHECK(index != -1, EC_DATA, "output closed");
	CHECK(index < maxSize, EC_DATA, "output: buffer overflow");
	buffer[index++] = b;
}

void MSOutputArray:: print () 
{
	PRINT("TODO");
}

} // namespace meanscript(core)

File: java/net/meanscript/MSOutputArray.java
package net.meanscript;
import net.meanscript.core.*;
import net.meanscript.java.*;
public class MSOutputArray extends MSOutputStream {
 byte  buffer [];
 int maxSize;
 int index;

public MSOutputArray ()
{
	maxSize = MSJava.globalConfig.outputArraySize;
	{ buffer = new byte[maxSize];  };
	index = 0;
}

@Override
public void close ()
{
	index = -1;
}

@Override
public void  writeByte (byte b) throws MException
{
	MSJava.assertion(index != -1, EC_DATA, "output closed");
	MSJava.assertion(index < maxSize, EC_DATA, "output: buffer overflow");
	buffer[index++] = b;
}

public void  print ()
{
	MSJava.printOut.print("TODO").endLine();
}

}
